crates/sui-json-rpc/src/api.rs

```diff
@@ -111,6 +112,19 @@ pub trait RpcReadApi {
         /// the ID of the queried object
         object_id: ObjectID,
     ) -> RpcResult<GetObjectDataResponse>;
+
+    /// Note there is no software-level guarantee/SLA that objects with past versions
+    /// can be retrieved with this API, even if the object and version exists/existed.
+    /// The result may vary across nodes depending on their pruning policies.
+    /// Return the object information for a specified version
+    #[method(name = "getPastObjectMaybe")]
+    async fn get_past_object_maybe(
```

Comment:
nit: not used to seeing "maybe" in method name lol, will `try_get_past_object` a better choice?